### PR TITLE
Implement the 'Add Specification' modal

### PR DESCRIPTION
**Pull request by calebj0seph**
Implement a modal to add specifications with a title, URL and description. Inputs are validated, with invalid inputs displaying with Material inline error messages. If a specification fails to add, an error modal appears.

output

### CODE REVIEW

File: src/view/AddSpecificationModal.tsx [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+                    <Button color="primary" onClick={() => this.onAddSpecification()}>
+                      Add
+                    </Button>
+                  )}
+                </div>
+              </div>
+            </Modal>
+            <Modal
+              open={this.state.showErrorModal}
+              onClose={() => this.hideFailureModal()}
```
Comment (PatrickShaw) [inline src/view/AddSpecificationModal.tsx]: You can't have `onClose={this.hideFailureModal}`

File: src/view/AddSpecificationModal.tsx [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+                    multiline={true}
+                    rowsMax={3}
+                    margin="normal"
+                  />
+                </div>
+                <div className={classes.buttonRow}>
+                  <Button
+                    color="primary"
+                    disabled={this.state.showProgressIndicator}
+                    onClick={() => this.closeModal()}
```
Comment (PatrickShaw) [inline src/view/AddSpecificationModal.tsx]: `onclick{this.closeModal}`?

**Comment by PatrickShaw** [comment]
Could probably greatly reduce verbosity with a bit of MobX. I'll give it a go now.

File: src/view/AddSpecificationModal.tsx (hunk 1 above)
Comment (calebj0seph) [reply src/view/AddSpecificationModal.tsx]: Will that work without autobind? Thought that `this` won't get passed into the method doing it that way.

**Comment by PatrickShaw** [comment]
@calebj0seph it didn't occur to me until just then but should Add Specification be at the top of the window? Otherwise people would have to scroll all the way down to the bottom of the page to add a spec (if they had lots of SDKs to manage)

**Comment by PatrickShaw** [comment]
Alright all good with me then.

**Pull request merged**